de.bund.bfr.knime.fsklab.deprecatednodes/src-1_7_2/de/bund/bfr/knime/fsklab/nodes/v1_7_2/joiner/JoinerNodeModel.java

```diff
@@ -336,14 +351,26 @@ private void creatRelationList(String relation, JoinerViewValue joinerProxyValue
           jR.setLanguage_written_in(sourceTargetRelation.getString("language_written_in"));
         }
         if (sourceTargetRelation.containsKey("sourceParam")) {
-          jR.setSourceParam(getObjectFromJson(sourceTargetRelation.get("sourceParam").toString(),
```

Comment:
I am lost here. So the whole *creatRelationList* loads a JsonString as a Javax JsonObject, processes directly simple properties (command and language written in) and for the source and target parameters they are converted to strings that are later deserialised with Jackson. Also, in the case that the content in the sourceParam or targetParam nodes is invalid this is using the whole raw content as the id of a new empty parameter (catch clause). Is this correct?

I would simplify this by data type binding with Jackson or simply reading the JSON string as a JsonTree and iterate like you do here. The nodes for the source and target parameters could be handled individually and deserialised to the Parameter classes. Do we have examples or tests with these join relations?

Reply:
I understand your concerns but There are two types of setting for the JoinRelations.json: 
one with the parameter ID only and the other (I think it is the older version) is hosting the whole parameter information.
I used Jackson to do the conversion into a Parameter Object and if it fails the catch will try to create a parameter with the existing parameter ID in the file.
please see: https://app.zenhub.com/files/241333403/fb898916-df34-48cd-ac4b-d5810b7be5f9/download
and https://seafile.bfr.berlin/f/6e84bfd933074036bd3e/